.github/workflows/base.yml

```diff
@@ -0,0 +1,73 @@
+name: Base CCCL workflow
+
+on:
+  workflow_call:
+    inputs:
+      matrix: # The matrix to use for this job
+        type: string
+        required: true
+      build_script: # The script to run for the build step
+        type: string
+        required: true
+      test_script: 
+        type: string
+        required: true
+
+jobs:
+  build:
+    if: github.repository == 'NVIDIA/cccl'
+    name: Build/CUDA ${{matrix.cuda}}/${{ matrix.compiler.name }}${{ matrix.compiler.version }}/C++${{matrix.std}}/${{matrix.cpu}}
+    strategy:
+      fail-fast: false
+      max-parallel: 8
+      matrix: ${{ fromJSON(inputs.matrix) }}
+    runs-on: linux-${{ matrix.cpu }}-cpu16
+    container: # Fetch the container image to use for the job
+      options: -u root
+      image: rapidsai/devcontainers:23.06-cpp-${{matrix.compiler.name}}${{matrix.compiler.version}}-cuda${{matrix.cuda}}-ubuntu${{ matrix.ubuntu }}
+    permissions: # Set permissions needed for sccache/AWS credentials
+      id-token: write
+      contents: read
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        persist-credentials: false
+    - name: Configure credentials and environment variables for sccache
+      uses: ./.github/actions/configure_cccl_sccache
+    - name: Run build script
+      run: |
+        chmod +x ${{ inputs.build_script }}
```

Comment:
You can run `chmod +x <script>` locally and commit the file changes to avoid having to include `chmod` in your workflows like this.